src/GraphQL/Internal/AST.hs

```diff
@@ -76,12 +130,12 @@ data OperationDefinition = Query    { getNode :: Node }
                          | Mutation { getNode :: Node }
                            deriving (Eq,Show)
 
-data Node = Node Name [VariableDefinition] [Directive] SelectionSet
+data Node = Node (Maybe Name) [VariableDefinition] [Directive] SelectionSet
             deriving (Eq,Show)
 
 -- XXX: Lots of things have names. Maybe we should define a typeclass for
```

Comment:
-1 from me I think. We already have some pretty weird compiler errors because we're writing generic code. Nailing certain types into place (instead of type classes or type families) does reduce the overall movement and makes  compiler errors more readable.